enhancements/observability/cluster-observability-operator-logging-stack.md

```diff
@@ -0,0 +1,405 @@
+
+Historically the OpenShift Logging product allows the creation **only** of a single logging stack per cluster. This design detail represents a choice to preserve resources on a cluster rather than a limitation. Having multiple end-to-end stacks is rather a very expensive solution and has been corroborated by different means:
+- Soft tenancy combined with full RBAC-based controls on logs. This design choice renders the need for multiple expensive log storage installations obsolete.
+- Using namespaced `ClusterLogForwarder` resources to offers users with configuration options on which workloads should be collected as well as how to filter logs.
+
+Therefore it is possible to allow creating multiple `LoggingStack` resources on the same cluster with the following workflow:
+1. Create main `LoggingStack` resource in `openshift-logging` namespace that reconciles the operator installation as well as the resources `ClusterLogging` and `Lokistack`.
+2. Create a `LoggingStack` resource on any other namespace can be rendered on a `ClusterLogForwarder` resource accounting for collecting the workload logs of that namespace and forwarding them to the `default` log store backed by the `LokiStack` resource.
+
+Limitations of the above worklow are:
+1. Opt-in to collect audit logs should be allowed only via the main `LoggingStack` resource in `openshift-logging`. Any other `LoggingStack` resource will be ignored on this matter.
+2. Using the API audit log filtering capabilities in `ClusterLogForwarder` (See [docs](clf-api-audit-log-filter)) is only possible in `openshift-logging`.
+
+In summary although the above workflow is feasible, the present proposal does **not** recommend any further implementation leaving users to manage their own `ClusterLogForwarder` resource per namespace if needed, because:
+1. The translation of per namespace `LoggingStack` resources to `ClusterLogForwarder` is of diminishing returns.
+2. More important the limitations of Server-Side-Apply on list items make several annotation makers on the `ClusterLogForwarder` resource mandatory (See later on [SSA-Limitations](#limitations-of-server-side-apply-on-openshift-logging-resources))
+
+#### Implications of Server-Side-Apply on OpenShift Logging resources
+
+The COO's design is centered around the Kubernetes Server-Side-Apply capability. In short it allows sharing resources between separate operator on the resource field level. However each OpenShift Logging resource has mandatory requirements on field when created. These fields will inevitably change management ownership from CLO/LO to the Cluster Obserbability Operator.
+
+In detail the following `ClusterLogForwarder` fields will be managed by COO exclusively without allowing user edits:
+
+```yaml
+apiVersion: observability.openshift.io/v1
+kind: ClusterLogForwarder
+metadata:
+  name: instance
+  namespace: openshift-logging
+spec:
+  serviceAccount:
+    name: cluster-logging
+  outputs:
+  - type: lokistack
```

Comment:
What secrets are needed and who is providing them?  5.x uses bearer token now?  Do we need to modify "lokistack" in the collector to expect a secret with a bearer token in lieu of handling "default" differently and implicitly mounting the SA token secret